langserve/server.py

```diff
@@ -866,6 +877,44 @@ async def playground(file_path: str, config_hash: str = "") -> Any:
             file_path,
         )
 
+    @app.post(namespace + "/feedback")
```

Comment:
We might need to put the endpoint behind a flag

```python

add_routes(...., accepts_feedback=True) # Would add the feedback
```

The issue is that if the server is publicly deployed and anonymous user that's both bored and malicious could leave incorrect feedback data.

We could do this if you like the solution:

1) Endpoint always appears
2) Client side error is raised when tracing is disabled OR when the endpoint does not accept feedback
3) We add additional metadata to the endpoint to be exposed via openapi specs that tells a js client whether the endpoint is enabled

Reply:
Making this endpoint optional to turn off makes a lot of sense, and agree with setting the default to `True`. A couple questions:

1) Why a client side error rather than a server side error that the client just translates? I feel like we could just throw a 400 if the feedback endpoint is called and it's off. Might be easier to deal with in the short term.
2) Should we do your third point above as a follow up?

Reply:
I realize we need some follow ups anyways on client handling. We'll need to figure out how to propagate the run_id as an output to the client in some way that doesn't interfere with it's output as a runnable.

Reply:
> Why a client side error rather than a server side error that the client just translates? I feel like we could just throw a 400 if the feedback endpoint is called and it's off. Might be easier to deal with in the short term.

Sorry by client side error i mean an error thrown on the server side with a 4xx status code. I think we're on the same page